app/controllers/planning_application/review_immunity_details_controller.rb

```diff
@@ -22,35 +23,50 @@ def edit
     end
 
     def update
-      @immunity_detail.assign_attributes(
-        review_status: status
-      )
+      @immunity_detail.assign_attributes(review_status:)
 
       respond_to do |format|
-        if @immunity_detail.update(immunity_detail_params)
+        if @immunity_detail.update(status: immunity_detail_status) &&
+           @review_immunity_detail.update(review_immunity_detail_params)
```

Comment:
Does this need to be in a transaction?
I guess the first update could still happen and be persisted